memory/level_up_manager.py

```diff
@@ -0,0 +1,123 @@
+                        return
+                    self.fields_base = self.memory.get_class_fields_base(singleton_ptr)
+
+                else:
+                    self._read_level_up_screen_active()
+                    if self._read_level_up_screen_active:
+                        self._read_current_level_up_upgrades()
+                        self._read_current_character()
+            except Exception as _e:  # noqa: F841
+                # logger.debug(f"Level Up Manager Reloading {type(_e)}")
+                self.__init__()
+
+    def _read_level_up_screen_active(self: Self) -> None:
+        ptr = self.memory.follow_pointer(self.base, [0x20, 0x0])
+        if ptr == 0x0:
+            self.level_up_screen_active = False
+        else:
+            self.level_up_screen_active = True
+
+    def _read_current_character(self: Self) -> None:
+        try:
+            definition_id_ptr = self.memory.follow_pointer(
+                self.base, [0x88, 0x88, 0x50, 0x58, 0x40, 0x0]
+            )
+            definition_id = self.memory.read_string(definition_id_ptr + 0x14, 8)
+            # LevelUpSceneController -> currentCharacter -> stateMachine -> currentState...
+            # -> player -> characterDefinitionId
+            definition_id_ptr = self.memory.follow_pointer(self.base, [0x88, 0x0])
+            character = PlayerPartyCharacter.parse_definition_id(definition_id)
+            self.current_character = character
+        except Exception:
+            self.current_character = PlayerPartyCharacter.NONE
+
+    def _read_current_level_up_upgrades(self: Self) -> None:
+        # LevelUpSceneController -> currentLevelUpUpgrades -> _items -> item[x]
+        self.memory.follow_pointer(self.base, [0xB0, 0x0])
+        try:
+            items = self.memory.follow_pointer(self.base, [0xB0, 0x10, 0x0])
+        except Exception:
+            self.current_level_up_upgrades = []
+            return
+
+        if items in {self.NULL_POINTER, self.ZERO_NULL_POINTER}:
+            self.current_level_up_upgrades = []
+            return
+        # Item is an array of pointers of size 0x08
+        upgrades = []
+        if items:
+            # Item objects are as follows:
+            # Items
+            #   - 0x18 - count
+            #   - 0x20 - Item[0]
+            #   - 0x28 - Item[1]
+            count = self.memory.read_int(items + 0x18)
+            address = self.ITEM_INDEX_0_ADDRESS
+            for _item in range(count):
+                item_ptr = self.memory.follow_pointer(items, [address, 0x0])
+                upgrade_type = LevelUpUpgradeType(self.memory.read_int(item_ptr + 0x38))
+                active_ptr = self.memory.follow_pointer(item_ptr, [0x50, 0x0])
+                active = not self.memory.read_bool(active_ptr + 0xA0)
+                upgrades.append(LevelUpUpgrade(upgrade_type, active))
+                address += self.ITEM_OBJECT_OFFSET
+        self.current_level_up_upgrades = upgrades
```

Comment:
shouldn't this be `self.current_upgrades`? Either way, we should not assign a value that isn't declared in `__init__()`

Reply:
Yes - will fix